File: project_euler/Problem003/Problem003.cpp
#include <iostream>
using namespace std;

class Problem003 {
public:
    void solution() {
        long long target = 600851475143L;
        while (!(target & 1)) {
            target >>= 1;
        }
        if (1 == target) {
            target = 2L;
        }

        long long factor = 3L;
        while (factor < target) {
            while (!(target % factor) && target != factor) {
                target /= factor;
            }
            factor += 2L;
        }
        cout << target << endl;
    }
};

File: project_euler/Problem003/Problem003.py
class Problem003:
    def solution(self) -> None:
        target = 600851475143
        while not (target & 1):
            target >>= 1
        if 1 == target:
            target = 2

        factor = 3
        while factor < target:
            while not target % factor and target != factor:
                target //= factor
            factor += 2
        print(target)
